### PR TITLE
thorvg: add version 0.15.1

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **thorvg/0.15.1**

#### Motivation
There are lots of improvements since 0.14.10.

#### Details
https://github.com/thorvg/thorvg/compare/v0.14.10...v0.15.1

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/thorvg/all/conanfile.py [hunk 1]
```diff
@@ -129,6 +129,9 @@ def requirements(self):
         if self.settings.os == "Linux":
             if self.options.with_engines in ["gl", "gl_beta"]:
                 self.requires("opengl/system")
+        if self.options.with_threads and Version(self.version) >= "0.15.0":
+            if self.options.get_safe("with_openmp") and self.settings.compiler in ["clang", "apple-clang"]:
```
Comment (franramirez688) [inline recipes/thorvg/all/conanfile.py:133]: @toge `with_openmp` option does not exist either in Conanfile or upstream.

Comment (franramirez688) [reply recipes/thorvg/all/conanfile.py:133]: As far as I saw, it only depends on the `threads` one: https://github.com/thorvg/thorvg/blob/b4238cd647f0d54fd4536c26cdf9d8ac99e0c747/src/renderer/sw_engine/meson.build#L23

File: recipes/thorvg/all/conanfile.py [hunk 2]
```diff
@@ -129,6 +129,8 @@ def requirements(self):
         if self.settings.os == "Linux":
             if self.options.with_engines in ["gl", "gl_beta"]:
                 self.requires("opengl/system")
+        if self.options.with_threads and Version(self.version) >= "0.15.0" and self.settings.compiler in ["clang", "apple-clang"]:
+            self.requires("llvm-openmp/17.0.4", transitive_headers=True, transitive_libs=True)
```
Comment (franramirez688) [inline recipes/thorvg/all/conanfile.py:133]: Do we have logs showing any failure if any of these or even both parameters are removed from here?

```suggestion
            self.requires("llvm-openmp/17.0.4")
```

File: recipes/thorvg/all/conanfile.py [hunk 3]
```diff
@@ -200,3 +202,15 @@ def package_info(self):
...
+            openmp_flags = []
+            if is_msvc(self):
+                openmp_flags = ["-openmp"]
+            elif self.settings.compiler == "gcc":
+                openmp_flags = ["-fopenmp"]
+            elif self.settings.compiler in ["clang", "apple-clang"]:
+                openmp_flags = ["-Xpreprocessor", "-fopenmp"]
+            self.cpp_info.cxxflags.extend(openmp_flags)
+            self.cpp_info.exelinkflags.extend(openmp_flags)
+            self.cpp_info.sharedlinkflags.extend(openmp_flags)
```
Comment (franramirez688) [inline recipes/thorvg/all/conanfile.py:216]: `llvm-openmp` recipe is already setting some flags in the cpp_info. I think that we can remove all of those:

```suggestion
```

Have you checked it locally?

**Review by franramirez688 — COMMENTED** [review]
Thanks for the PR @toge 👏 

Looking great! Only a couple of comments before moving forward 😉

File: recipes/thorvg/all/conanfile.py (hunk 3 above)
Comment (toge) [reply recipes/thorvg/all/conanfile.py:216]: @franramirez688 
Thank you for your review!
I try to remove all codes, but I met compilation errors:
```
[100%] Linking CXX executable test_package
/usr/bin/ld: lib/libthorvg.a(renderer_sw_engine_tvgSwRenderer.cpp.o): in function `tvg::SwRenderer::init()':
tvgSwRenderer.cpp:(.text+0x217c): undefined reference to `omp_set_num_threads'
/usr/bin/ld: lib/libthorvg.a(renderer_sw_engine_tvgSwRaster.cpp.o): in function `rasterXYFlip(unsigned int*, unsigned int*, int, int, int, SwBBox const&, bool) [clone ._omp_fn.0]':
tvgSwRaster.cpp:(.text+0x244d): undefined reference to `omp_get_num_threads'
/usr/bin/ld: tvgSwRaster.cpp:(.text+0x2454): undefined reference to `omp_get_thread_num'
/usr/bin/ld: lib/libthorvg.a(renderer_sw_engine_tvgSwRaster.cpp.o): in function `rasterXYFlip(unsigned int*, unsigned int*, int, int, int, SwBBox const&, bool)':
tvgSwRaster.cpp:(.text+0xdefd): undefined reference to `GOMP_parallel'
/usr/bin/ld: lib/libthorvg.a(renderer_sw_engine_tvgSwPostEffect.cpp.o): in function `_gaussianBlur(unsigned char*, unsigned char*, int, int, int, SwBBox const&, int, int, bool) [clone ._omp_fn.0]':
tvgSwPostEffect.cpp:(.text+0x15): undefined reference to `omp_get_num_threads'
/usr/bin/ld: tvgSwPostEffect.cpp:(.text+0x1c): undefined reference to `omp_get_thread_num'
/usr/bin/ld: lib/libthorvg.a(renderer_sw_engine_tvgSwPostEffect.cpp.o): in function `effectGaussianBlur(SwImage&, SwImage&, SwBBox const&, tvg::RenderEffectGaussian const*)':
tvgSwPostEffect.cpp:(.text+0x1094): undefined reference to `GOMP_parallel'
/usr/bin/ld: tvgSwPostEffect.cpp:(.text+0x1152): undefined reference to `GOMP_parallel'
/usr/bin/ld: tvgSwPostEffect.cpp:(.text+0x1258): undefined reference to `GOMP_parallel'
/usr/bin/ld: tvgSwPostEffect.cpp:(.text+0x1412): undefined reference to `GOMP_parallel'
/usr/bin/ld: tvgSwPostEffect.cpp:(.text+0x1518): undefined reference to `GOMP_parallel'
/usr/bin/ld: lib/libthorvg.a(renderer_sw_engine_tvgSwPostEffect.cpp.o):tvgSwPostEffect.cpp:(.text+0x15df): more undefined references to `GOMP_parallel' follow
collect2: error: ld returned 1 exit status
gmake[2]: *** [CMakeFiles/test_package.dir/build.make:107: test_package] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/test_package.dir/all] Error 2
gmake: *** [Makefile:91: all] Error 2
```

It is caused by the lack of `execlinkflags`.
So I am going to modify this recipe as followings:
```python
if self.options.with_threads and Version(self.version) >= "0.15.0":
    openmp_flags = []
    if is_msvc(self):
        openmp_flags = ["-openmp"]
    elif self.settings.compiler == "gcc":
        openmp_flags = ["-fopenmp"]
    elif self.settings.compiler in ["clang", "apple-clang"]:
        openmp_flags = ["-Xpreprocessor", "-fopenmp"]
    self.cpp_info.exelinkflags.extend(openmp_flags)
```

**Review by franramirez688 — APPROVED** [review]
Hi @toge 

Sorry for the delay, but I paid special attention to the `openmp` part. After discussing it with @jcar87, we decided to drop it to avoid extra complexity because its benefits are minimal for now.

Again, thanks a lot for your efforts! From my side, the PR is ready 👏

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**